l10n_it_amount_to_text/models/res_currency.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Ecosoft Co., Ltd (http://ecosoft.co.th)
+# Copyright 2022 Sergio Zanchetta (Associazione PNLUG - Gruppo Odoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from num2words import num2words
+
+from odoo import models, tools
+
+
+class Currency(models.Model):
+    _inherit = "res.currency"
+
+    def amount_to_text(self, amount):
+        self.ensure_one()
+        lang = tools.get_lang(self.env)
+        lang_code = lang.code
+        if lang_code != "it_IT":
+            return super().amount_to_text(amount)
+
+        def _num2words(number, lang):
+            try:
+                return num2words(number, lang=lang)
+            except NotImplementedError:
+                return num2words(number, lang="en")
+
+        formatted_amount = "%.{}f".format(self.decimal_places) % amount
+        parts = formatted_amount.partition(".")
+        integer_value = int(parts[0])
+        fractional_value = int(parts[2] or 0)
+
+        # use num2words native 'currency' option to handle exceptions.
+        # eg. 'uno euro' -> 'un euro', 'uno centesimi' -> 'un centesimo'"""
+        try:
+            return num2words(
+                formatted_amount, to="currency", lang=lang.iso_code, currency=self.name
+            )
+        except NotImplementedError:
+            amount_words = tools.ustr("{amt_value} {amt_word}").format(
```

Comment:
A quanto ho capito tutto l'override di questo metodo serve per passare i parametri `currency` e `to_currency` a `num2words`, se in questo ramo di codice non viene fatto forse si può semplicemente chiamare `super`?

Mi pare che l'unica differenza con `super` sia il `lower` applicato poco sotto, forse si potrebbe applicare al risultato di `super`? Che ne pensi?

Reply:
Quella attuale mi sembra una soluzione più pulita perché per cerca di ridurre le operazioni necessarie per ottenere il risultato.
Nel caso di un'altra chiamata a `super` ad esempio verrebbe applicato un `title` seguito poi da un `lower`.